src/client/shader.cpp

```diff
@@ -790,7 +833,13 @@ ShaderInfo ShaderSource::generateShader(const std::string &name,
 		dumpShaderProgram(warningstream, "Vertex", vertex_shader);
 		dumpShaderProgram(warningstream, "Fragment", fragment_shader);
 		dumpShaderProgram(warningstream, "Geometry", geometry_shader);
-		return shaderinfo;
+		throw ShaderException(
+			"Failed to compile the \"" + name + "\" shader.\n" +
+			shader_compilation_errors +
+			"Check debug.txt for details.");
+	} else if (!shader_compilation_errors.empty()) {
+		// Don't suppress any error messages.
+		errorstream << shader_compilation_errors;
```

Comment:
`std::flush` to ensure they're not written seconds later with the next log call.

Reply:
This only matters if we include 87c6497.